nasa-gibs/worldview
# WV-3235-Support Time-Limited Display of TEMPO Imagery

## Description

>GITC needs to have the ability to display products that have irregular times, like TEMPO. ASDC wants to show L2 and L3 TEMPO imagery, which have variable time duration as well as variable time gaps between imagery. 

>The current GIBS behavior has an image remain in the map viewer until the next timestamp with imagery. However, each day, TEMPO scans stop when daylight ends and does not begin again until daylight the next day. As a result, the last image of the day remains in the viewer for hours past the end timestamp of the image. In addition, there are gaps in data coverage during the day (e.g., TEMPO special operations, satellite maneuvers, errors) that result in an image remaining for an hour(s) past the end time of the image. Due to the high temporal resolution of the data, having the imagery remain in the viewer for longer than the scan time may cause confusion in users and has been expressed as a concern by the TEMPO science team. 

>The ASDC would like to have the TEMPO imagery only be in the map viewer for the time length of the imagery (i.e., limit to start/end time of imagery), regardless of the timestep within the imagery (e.g., some images are for 6 min, some 40 min, some 60 min), and regardless of how long it is until the next image occurs (e.g., if the image starts the following minute, not for a couple of hours, or not until the following day). 

## How To Test

1. `git checkout WV-3235-time-limit-tempo`
2. `npm ci && npm run build && npm run watch`
3. Open this [link](http://localhost:3000/?v=-178.51395268621218,-22.136441945375765,-5.2571551031306,77.42605805462424&z=4&ics=true&ici=5&icd=6&l=TEMPO_L2_Cloud_Cloud_Pressure_Total_Granule(count=1)&lg=true&t=2024-07-16-T17%3A48%3A00Z).
4. The granules should disappear when selected date/time is outside of the cmr data availability range.

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

## Merging

Please use the `squash and merge` commit method unless each commit in your branch is vital to the commit history of main.

@nasa-gibs/worldview
